### PR TITLE
US884179: Change API logger level to warn and don't log stack trace

### PR DESCRIPTION
**Pull request by raghavendran-gopalakrishnan**
https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=884179

### CODE REVIEW

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 1]
```diff
@@ -179,13 +179,13 @@ public ResponseEntity<BatchList> getBatches(
...
-            LOGGER.error("Invalid X-TENANT-ID ", ex);
+            LOGGER.warn("Invalid X-TENANT-ID {}", X_TENANT_ID);
             throw new WebMvcHandledRuntimeException(HttpStatus.BAD_REQUEST, ex.getMessage());
         } catch (final InvalidBatchIdException ex) {
-            LOGGER.error("Invalid batchId ", ex);
+            LOGGER.warn("Invalid batchId {}", ex.getMessage());
             throw new WebMvcHandledRuntimeException(HttpStatus.BAD_REQUEST, ex.getMessage());
         } catch (final StagingException ex) {
-            LOGGER.error("Error in getBatches ", ex);
+            LOGGER.warn("Error in getBatches {}", ex.getMessage());
```
Comment (dermot-hardy) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: Are we sure the stack trace would not be useful in this case?

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 2]
```diff
@@ -94,15 +94,15 @@ public ResponseEntity<Void> createOrReplaceBatch(final String X_TENANT_ID, final
             try {
                 fileItemIterator = fileUpload.getItemIterator(request);
             } catch (final IOException ex) {
-                LOGGER.error("Error getting FileItemIterator", ex);
+                LOGGER.warn("Error getting FileItemIterator: {}", ex.getMessage());
```
Comment (dermot-hardy) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: I feel like the stack trace might be useful in this case.

**Review by dermot-hardy — CHANGES_REQUESTED** [review]
You may have applied this too widely.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java (hunk 2 above)
Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: Ya it makes sense.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java (hunk 1 above)
Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: I think it would be. Putting it back.

Comment (andyreidz) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: This doesn't seem right, Dermot is asking for stack trace, I don't believe the message contains that.

Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: Ya I meant I have reverted my change of getting message from exception object. Stack trace would be printed like before.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 3]
```diff
@@ -140,19 +140,19 @@ public ResponseEntity<BatchStatusResponse> getBatchStatus(final String X_TENANT_
...
-            LOGGER.error("Invalid Batch Id", ex);
+            LOGGER.warn("Invalid Batch Id {}", batchId);
             throw new WebMvcHandledRuntimeException(HttpStatus.BAD_REQUEST, ex.getMessage());
         } catch (final BatchNotFoundException ex) {
-            LOGGER.error("Batch not found", ex);
+            LOGGER.warn("Batch {} not found", batchId);
             throw new WebMvcHandledRuntimeException(HttpStatus.NOT_FOUND, ex.getMessage());
         } catch (final StagingException ex) {
-            LOGGER.error("Internal server error", ex);
+            LOGGER.warn("Internal server error: {}", ex.getMessage());
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: This looks like a case were we should pass ex?

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 4]
```diff
@@ -140,19 +140,19 @@ public ResponseEntity<BatchStatusResponse> getBatchStatus(final String X_TENANT_
...
             throw new WebMvcHandledRuntimeException(HttpStatus.INTERNAL_SERVER_ERROR, ex.getMessage());
         } catch (final InterruptedException ex) {
-            LOGGER.error("Service Unavailable", ex);
+            LOGGER.warn("Service Unavailable: {}", ex.getMessage());
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: This looks like a case were we should pass ex?

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java [hunk 5]
```diff
@@ -179,10 +179,10 @@ public ResponseEntity<BatchList> getBatches(
             batchList.entries(batchFiles);
             return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(batchList);
         } catch (final InvalidTenantIdException ex) {
-            LOGGER.error("Invalid X-TENANT-ID ", ex);
+            LOGGER.warn("Invalid X-TENANT-ID {}", X_TENANT_ID);
             throw new WebMvcHandledRuntimeException(HttpStatus.BAD_REQUEST, ex.getMessage());
         } catch (final InvalidBatchIdException ex) {
-            LOGGER.error("Invalid batchId ", ex);
+            LOGGER.warn("Invalid batchId {}", ex.getMessage());
```
Comment (andyreidz) [inline staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: Above I see we supply batchId in what looks like a similar case?
image

Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: Yes changed to `from`.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java (hunk 4 above)
Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: addressed.

File: staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java (hunk 3 above)
Comment (raghavgopal) [reply staging-service/src/main/java/com/github/cafdataprocessing/services/staging/StagingController.java]: addressed

**Review by rorytorneymf — APPROVED** [review]

**Pull request merged**